docs/blog/2025-05-02-noir-and-zkemail-mobile-packages.md

```diff
@@ -0,0 +1,107 @@
+
+1. **Rust Crate** – we expose `proveZkemail` & `verifyZkemail` around Barretenberg backend.
+2. **UniFFI** consumes a `.udl` interface for bindings.  
+3. **Build Tools** – `mopro build` creates a static **`.a`** for iOS and a dynamic **`.so`** for Android; CocoaPods, SwiftPM, Gradle, Pub and npm publish the artifacts.  
+4. **Your App** – calls the generated functions exactly like platform code; no manual JNI or Objective-C shims required.
+
+### Example Usage (Swift)
+
+```swift
+import ZKEmailSwift
+let inputs = ["header_storage": ["..."]]
+let proof = try proveZkemail(srsPath: "/path/to/srs", inputs:  inputs)
+let isValid = try verifyZkemail(srsPath: "/path/to/srs", proof:   proof)
+```
+*Functions `proveZkemail` and `verifyZkemail` come straight from* `zkemail-swift-package`*.* 
+
+### Example Usage (Kotlin)
+
+```kotlin
+import uniffi.mopro.*
+val inputs = mapOf("header_storage" to listOf("..."))
+val proof = proveZkemail(srsPath, inputs)
+val isValid = verifyZkemail(srsPath, proof)
+```
+*These top-level functions are generated by UniFFI and exposed in* `zkemail-kotlin-package`.
+
+## Current Benchmarks
+
+All numbers were captured on an **Apple M3 8-core** device and were built in **release** mode. The circuit is the header-only proof from [zkemail.nr_header_demo](https://github.com/Mach-34/zkemail.nr_header_demo); proof size is a flat 28 KB across platforms .
+
+| **Operation**    | **iOS (ms)** | **Android (ms)** |
+|------------------|-------------:|-----------------:|
+| Proof Generation | **1,309**    | 3,826 |
+| Verification     | **962**      | 2,857 |
+
+<p align="center">
+    <table>
+    <tr>
+        <td align="center">
+        <a href="/img/zkemail-flutter-app-ios.png" target="_blank" rel="noopener noreferrer">
+            <img src="/img/zkemail-flutter-app-ios.png" alt="iOS zkEmail App Example" width="300"/>
+        </a>
+        <br />
+        <sub><b>iOS</b></sub>
+        </td>
+        <td align="center">
+        <a href="/img/zkemail-flutter-app-android.png" target="_blank" rel="noopener noreferrer">
+            <img src="/img/zkemail-flutter-app-android.png" alt="Android zkEmail App Example" width="300"/>
+        </a>
+        <br />
+        <sub><b>Android</b></sub>
+        </td>
+    </tr>
+    </table>
+    <p align="center">
+        <em>Flutter App for iOS & Android zkEmail Example</em>
+    </p>
+</p>
+
+## Under The Hood: Managing Static Library Size On iOS
+
+### Why Static Linking?
+
+UniFFI exports Swift bindings as a **static** archive (`libmoproiosbindings.a`).  
+Static linking guarantees that every Rust symbol is available at link-time and avoids Objective-C trampolines, simplifying Xcode integration. The trade-off is that all Rust dependencies—including Barretenberg, `rayon`, and big-integer math—end up inside the archive.
+
+### Baseline Size
+
+A full build produces an archive of **≈ 153 MB**.  
+While acceptable for some package management platforms, the raw size slows down:
```

Comment:
we can mention that github does not accept and it causes the package failed to use when importing from github